### PR TITLE
Uhuruhashimoto/distance conversion

### PR DESCRIPTION
**Pull request by uhuruhashimoto**
## Purpose
Centralize GPS conversion functions into a module/toolbox.

## Implementation
Distance conversion utilities are spread between `aux_functions` and `constructors`; make an upper-level data conversion module that both can pull from. Module only exposes *conversion* functions; it does not generate random data or expose conversion constants (`distance_constants`).

Note: also add a `gps floating point` -> `proto field` function that mocks can call.

## Tests
Unit tests updated and pass.

### CODE REVIEW

**Review by barkincavdaroglu — COMMENTED** [review]
Since the conversion will only be used during centralization of graph, doesn't it make more sense to put this in init (or whatever directory will be responsible for constructing graph from messages)?

**Comment by uhuruhashimoto** [comment]
> Since the conversion will only be used during centralization of graph, doesn't it make more sense to put this in init (or whatever directory will be responsible for constructing graph from messages)?

TLDR: Yes, except for a single little wrench in the works: `take_snapshot`. 

Currently, both `constructor` and `aux` modules use distance conversion: 
- `take_snapshot` uses the `total_distance` function
- `init_graph` uses the `get_distance` wrapper for the same function

This happens because we want to use the Haversine distance to (1) sort nodes in a snapshot, and (2) use the distance between nodes to add edges when we mock edges for our existing map of node locations.

My solution was to make a top-level `data_conversion` module that contains *only converters*, and that can be used by both modules.

@barkincavdaroglu  Mind explaining what `take_snapshot` does and why it needs to rank nodes by distance?

**Comment by barkincavdaroglu** [comment]
> > Since the conversion will only be used during centralization of graph, doesn't it make more sense to put this in init (or whatever directory will be responsible for constructing graph from messages)?
> 
> TLDR: Yes, except for `take_snapshot`, which I don't understand enough to mess with.
> 
> Currently, both `constructor` and `aux` modules use distance conversion:
> 
> * `take_snapshot` uses the `total_distance` function
> * `init_graph` uses the `get_distance` wrapper for the same function
> 
> This happens because we want to use the Haversine distance to (1) sort nodes in a snapshot, and (2) use the distance between nodes to add edges when we mock edges for our existing map of node locations.
> 
> My solution was to make a top-level `data_conversion` module that contains _only converters_, and that can be used by both modules.
> 
> @barkincavdaroglu Mind explaining what `take_snapshot` does and why it needs to rank nodes by distance?

#97 and #94

**Review by barkincavdaroglu — APPROVED** [review]
